robot_sf/data_analysis/plot_dataset.py

```
@@ -0,0 +1,134 @@
+    """
+    Plot all NPC pedestrian velocities from the given JSON file.
+    Based on the actions of the npc pedestrians.
+
+    Args: raw (bool): If raw is True, dont use the applied scaling factor (for visual purpose).
+    """
+
+    # Extract pedestrian actions
+    ped_actions = extract_key_from_json(filename, "ped_actions")
+
+    velocity_list = []
+    for timestep, actions in enumerate(ped_actions):
```

Ruff (0.8.2)

Loop control variable `timestep` not used within loop body

(B007)